### PR TITLE
UI iteration 2

### PR DESCRIPTION
**Pull request by nostrbuddha**
- MyTrades UI (3rd tab) with Presenter, model, repository and mock data

 - Market screen: CurrencyListing screen with Presenter, model, repository and mock data

 - Market screen: OfferListing screen

These are basic UI and miss functionality like search, filters, switch between buy / sell. But I guess it's good enough to pull real data from bisq libs and populate in view for testing.

### CODE REVIEW

**Review by HenrikJannsen — CHANGES_REQUESTED** [review]

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/OffersListScreen.kt
Comment (rodvar) [inline bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/OffersListScreen.kt:73]: Are we dropping the dependency on the translator or we forgot to add it here? 

I'm ok if we hardcode in english for the MVP just want to make sure is a conscious decision.

**Review by rodvar — CHANGES_REQUESTED** [review]
outstanding work buddha :clap: 

Besides my code review comments, here is my functional review

 1. This PR is the perfect opportunity to implement back button navigation. At the moment, once you get to the Offers screen the only way to get out is by swiping left (which is great), but we need also the standard visual left arrow on the top left (next to the title) as it has become industry std in both iOS and Android these days. If you think this is a big deal please let me know and we can do it in a separate feature.
  2. Search bar: I can see that is just mocked at the moment, do we have a separate issue to work this one out? I'll refrain from commenting on that one for now (unless you ask me to do it now)
  
  That's it for this first pass, let me know if you have any questions! :muscle:

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/OffersListScreen.kt
Comment (HenrikJannsen) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/OffersListScreen.kt:73]: Do we need the dialogue? In Bisq Easy we dont have it but open the take offer screens (select the available options or get review screen if nothing to select). Though we have plenty of warnings with the reputation based amount limits. That will be painful to replicate ;-(.

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt
Comment (rodvar) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt:30]: General rule in mobile is don't show anything that the user cannot interact with. If there are no offers it should be hidden, but as you said let's leave this for fine tuning! ;)

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/OffersListScreen.kt
Comment (rodvar) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/OffersListScreen.kt:73]: great question. I think we should just go to the trade details and the user can decide if they want to take it there?
For now we can just log something on click to indicate the todo

good that he developed a dialog though, I'm sure it will be handy somewhere :)  ("Confirm Trade paid? Y/N e.g.")

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt
Comment (HenrikJannsen) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt:30]: I was wrong its not 150 markets as we filter for those where we have price, i guess about 30- 40 then... added the filter in my new branch. But still agree we might find a better UX to not show empty markets in a fine tune phase.

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/trades/MyTradesScreen.kt
Comment (nostrbuddha) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/trades/MyTradesScreen.kt:37]: Weird. I just re-checked in my project. And I see only one instance of `MyTradesScreen` in this commit: `f52ac0623e`

Can you please tell me the other instance where this function is defined

**Comment by nostrbuddha** [comment]
> outstanding work buddha 👏
> 
> Besides my code review comments, here is my functional review
> 
>     1. This PR is the perfect opportunity to implement back button navigation. At the moment, once you get to the Offers screen the only way to get out is by swiping left (which is great), but we need also the standard visual left arrow on the top left (next to the title) as it has become industry std in both iOS and Android these days. If you think this is a big deal please let me know and we can do it in a separate feature.
> 
>     2. Search bar: I can see that is just mocked at the moment, do we have a separate issue to work this one out? I'll refrain from commenting on that one for now (unless you ask me to do it now)
> 
> 
> That's it for this first pass, let me know if you have any questions! 💪

1. Yeah. Missed that one. Guess Henrick already implemented the back button. And we discussed (in matrix) a way of improving and generalising it inside the TopBar. I will do that now.

2. I will implement search functionality as well in this PR now.

File: bisqapps/shared/domain/src/commonMain/kotlin/network/bisq/mobile/domain/data/model/MyTrades.kt
Comment (nostrbuddha) [reply bisqapps/shared/domain/src/commonMain/kotlin/network/bisq/mobile/domain/data/model/MyTrades.kt:3]: My bad. Will update the comment!

I meant to refer Bisq2 for the data model structure of Offers and replicate the same here

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/CurrencyProfileCard.kt
Comment (nostrbuddha) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/CurrencyProfileCard.kt:53]: Yes. Will make sure this never happens again!

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/molecules/BisqDialog.kt
Comment (nostrbuddha) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/molecules/BisqDialog.kt:21]: oh okay. I ll create pixel 6a emulator and start testing in it.

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt
Comment (nostrbuddha) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt:30]: So there is filter option in the CurrencyListing (yet to be implemented).

In Bisq2 desktop, by default it shows all currencies.
But in mobile, we can set the `showmarkets` filter to `only with offers` by default.

Screenshot 2024-11-26 at 10 28 28 AM

and right now, I will hardcode all 150+ currnecies that are shown in Bisq2 desktop app.

Comment (nostrbuddha) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt:30]: For loader, I was thinking of 2 scenarios.

Scenario 1:
1. A listing page, where all data needs to be coming from an api call.
2. In this case, we can just pass `isLoading = true` and `loadingWidget = {Component}` to The BisqScaffold/BisqLayout classes and they take care of rendering the given component in the mid of the screen till the data arrives.

Scenario 2:
1. A form page, where all controls can be rendered initially with empty fields,
2. An overlay loader can be shown with blurred background so the form/controls are visible in the background, but cannot be interacted with
3.  Once the data arrives, overlay loader is removed and the form is populated with actual data.

Even to achieve this, we set only `isLoading = true` to The BisqScaffold/BisqLayout classes. Without setting `loadingWidget` when isLoading is set to true, they show the overlay loader.

Is this understandable / sounds good? Please let me know.

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/OffersListScreen.kt
Comment (nostrbuddha) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/OffersListScreen.kt:73]: > Are we dropping the dependency on the translator or we forgot to add it here?
> 
> I'm ok if we hardcode in english for the MVP just want to make sure is a conscious decision.

My bad!! Missed it. I will add the translation strings everywhere

Comment (nostrbuddha) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/OffersListScreen.kt:73]: Yes. Will remove the Dialog from here

**Comment by HenrikJannsen** [comment]
>     2. I will implement search functionality as well in this PR now.

You can leave that to do after the domain integration as it will likely be different than. I have already a filtered and sorted list and it will be easy to add that predicate.

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt
Comment (HenrikJannsen) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt:30]: I would suggest to leave the loading topic for later. For the markets / currency list I doubt it will be needed (at node its instant, at client we can hard code data as its static data).
Of other screens like offer list, the list gets automatically updated with data, so data stream in. That is also the case in Desktop where at first start it takes a bit until the offers arrive and then fill up the list as they arrive.

But I think different user cases will have diff. context here, which will be easier to analyze once the core functionality is implemented.

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/trades/MyTradesScreen.kt
Comment (HenrikJannsen) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/trades/MyTradesScreen.kt:37]: Ah just checked that commit and don't see it anymore. Maybe it was a copy/past error on my branch based on yours...

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt
Comment (HenrikJannsen) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt:30]: > So there is filter option

I would keep that also for later, it at all (IMO we should try to keep mobile to the most minimal feature set as possible).

> I will hardcode all 150+ currnecies that are shown in Bisq2 desktop app.

No need. I have it already connected with real data.

Comment (nostrbuddha) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt:30]: okay

**Comment by nostrbuddha** [comment]
@rodvar Made some updates now. Mostly minor:
 - dead code removal, better comments, translation keys/values

 - Adding all currencies are done by Henrik in his commits
 - For Search, I will work after his merge.
 - For small screen layout issues, I will create another issue and solve it.
 - Back button generalisation - I will do once Henrik's code is merged
 - Loader - Not for now.

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/CurrencyProfileCard.kt
Comment (rodvar) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/CurrencyProfileCard.kt]: haha no big deal but whenever we are sure let's keep it clean :)

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/layout/StaticLayout.kt
Comment (rodvar) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/layout/StaticLayout.kt:14]: @nostrbuddha yes please - let's create a separate issue so we don't loose track of this and eventually tackle it :muscle:

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt
Comment (rodvar) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt:30]: @nostrbuddha your comment on the scenarios can go straight into the description of a new "Loading Spinners" UI issue ;)

**Comment by rodvar** [comment]
> @rodvar Made some updates now. Mostly minor:
> 
> * dead code removal, better comments, translation keys/values
> 
> Others:
> 
> * Adding all currencies are done by Henrik in his commits
> * For Search, I will work after his merge, as suggested by Henrik
> * For small screen layout issues, I will create another issue and solve it.
> * Back button generalisation - I will do once Henrik's code is merged
> * Loader - Not for now.

excellenty buddha, I'll review and try to get this merge as is. Please create those issues so we don't look this track (or let me know if you want me to create them for you).

Cheers!

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/trades/MyTradesScreen.kt
Comment (rodvar) [inline bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/trades/MyTradesScreen.kt:45]: Let's avoid LaunchedEffect in favour of our new composable `RememberPresenterLifecycle` -> I'll update this PR so we can merge it now

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/trades/MyTradesScreen.kt
Comment (rodvar) [inline bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/trades/MyTradesScreen.kt:49]: good reactive ui programming

**Review by rodvar — APPROVED** [review]
cool buddha, just gonna fix the comment I've added now and merge it.

We need to create some GH issues to not loose track of the things we have decided to work on later in the context of this PR. 

I'll add one more here:

 - on iOS: standard left-to-right swipe is not going backwards on the view stack (in android back swipe which is the opposite direction works fine)
 
I'll let you decide if you prefer to create the issues yourself, happy to help you if you want to just keep the coding momentum - just please let me know :pray:

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/layout/StaticLayout.kt
Comment (nostrbuddha) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/layout/StaticLayout.kt:14]: Created one @ https://github.com/bisq-network/bisq-mobile/issues/79

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/molecules/BisqDialog.kt
Comment (nostrbuddha) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/molecules/BisqDialog.kt:21]: Yes. Will cover this in https://github.com/bisq-network/bisq-mobile/issues/79 as well

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt
Comment (nostrbuddha) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt:30]: Created at https://github.com/bisq-network/bisq-mobile/issues/80

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/trades/MyTradesScreen.kt
Comment (nostrbuddha) [reply bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/trades/MyTradesScreen.kt]: Similar to creating custom hooks in React - love it!
Thanks for showing this approach. 😊

**Pull request merged**
